charger/vestel.go

```diff
@@ -121,7 +123,12 @@ func NewVestel(ctx context.Context, uri string, id uint8) (api.Charger, error) {
 		return nil, fmt.Errorf("failed to read firmware version: %w", err)
 	}
 
-	fw := strings.TrimPrefix(bytesAsString(b), "v")
+	utf16BytesToString := func(b []byte) string {
+		s, _ := unicode.UTF16(unicode.BigEndian, unicode.IgnoreBOM).NewDecoder().String(string(b))
+		return s
+	}
+
+	fw, _, _ := strings.Cut(strings.TrimPrefix(utf16BytesToString(bytes.TrimRight(b, "\x00")), "v"), "-")
```

Comment:
Also entweder auf die Funktionsdeklaration verzichten oder (vllt. besser) den ganzen Kram in eine neue Funktion verschieben?

Reply:
charger/helper.go würde ich spontan sagen. Da ist auch schon `bytesAsString()` drin.

Reply:
Also 

    utf16BytesAsString